### PR TITLE
Make sure transform propagate always takes the most replayed replay.

### PR DESCRIPTION
**Pull request by csarofeen**

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/transform_replay.cpp [hunk 1]
```diff
@@ -718,17 +718,38 @@ bool TransformPropagator::replayPasC(
...
+      producer_root.end(),
+      [&dep_vals_set](IterDomain* root_id) {
+        return dep_vals_set.find(root_id) != dep_vals_set.end();
+      });
+
   if (replayed_pos.find(producer_tv) != replayed_pos.end()) {
-    if (producerAsC.second <= replayed_pos.at(producer_tv)) {
+    if (n_transformed_root_dims <= n_replayed_root_dims.at(producer_tv) ||
+        (n_transformed_root_dims == n_replayed_root_dims.at(producer_tv) &&
+         replayed_producer.second <= replayed_pos.at(producer_tv))) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/transform_replay.cpp]: If I'm reading this correctly, the rhs of `||` seems redundant. To be true, `n_transformed_root_dims == n_replayed_root_dims.at(producer_tv)` must hold, but then that's covered by the lhs condition of `||`.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/transform_replay.cpp]: ~Do you have an example fusion for which this change would fix the propagation?~

Found one in the header file.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/transform_replay.cpp]: Suppose `n_transformed_root_dims == n_replayed_root_dims.at(producer_tv)`. Wouldn't we want to return true when `replayed_producer.second > replayed_pos.at(producer_tv)`? Not sure if such a case could actually happen, though.

**Comment by rdspring1** [comment]
Can you share the original batch norm kernel, which motivated this PR?

File: torch/csrc/jit/codegen/cuda/transform_replay.cpp (hunk 1 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/transform_replay.cpp]: I think this should be:
```
n_transformed_root_dims < n_replayed_root_dims.at(producer_tv) ||
        (n_transformed_root_dims == n_replayed_root_dims.at(producer_tv) &&
         replayed_producer.second <= replayed_pos.at(producer_tv))
```
Just forgot to change the first section without the second. I don't know if it's necessary, but seemed harmless.

**Review by naoyam — APPROVED** [review]

**Comment by csarofeen** [comment]
It's one of the new TIMM C++ benchmarks. Once all the PRs are in just revert the PR and it should pop up as a scheduling failure. Sorry, I don't still have the debug fusion ir output otherwise would share that.

**Pull request merged**